genesis/sensors/base_sensor.py

```diff
@@ -162,31 +174,104 @@ def read_ground_truth(self, envs_idx: List[int] | None = None):
+
+    def pause_recording(self):
+        """
+        Pause recording the sensor data.
+        """
+        if self._data_recorder is None:
+            gs.raise_exception("Sensor data recorder not found. Call `add_recorder()` first.")
+        self._data_recorder.pause()
+
+    def stop_recording(self):
+        """
+        Stop recording the sensor data.
+        """
+        if self._data_recorder is None:
+            gs.raise_exception("Sensor data recorder not found. Call `add_recorder()` first.")
+        self._data_recorder.stop()
+
+    @property
+    def is_built(self) -> bool:
+        return self._manager._sim._scene._is_built
+
+    # =============================== private shared methods ===============================
+
     @classmethod
     def _apply_delay_to_shared_cache(
-        self, shared_metadata: SharedSensorMetadata, shared_cache: torch.Tensor, buffered_data: "TensorRingBuffer"
+        cls,
+        shared_metadata: SharedSensorMetadata,
+        shared_cache: torch.Tensor,
+        buffered_data: "TensorRingBuffer",
+        jitter: torch.Tensor | None = None,
+        interpolate_for_delay: list[bool] | None = None,
     ):
         """
         Applies the read delay to the shared cache tensor by copying the buffered data at the appropriate index.
+
+        NOTE: This is a helper method for `update_shared_cache()` to apply the delay to the shared cache tensor.
+
+        Parameters
+        ----------
+        shared_metadata : SharedSensorMetadata
+            The shared metadata for the sensor.
+        shared_cache : torch.Tensor
+            The shared cache tensor.
+        buffered_data : TensorRingBuffer
+            The buffered data tensor.
+        jitter : torch.Tensor | None
+            The jitter in steps before the sensor data is read.
+        interpolate_for_delay : list[bool] | None
+            Whether to interpolate the sensor data for the read delay.
         """
-        idx = 0
-        for tensor_size, read_delay_step in zip(shared_metadata.cache_sizes, shared_metadata.read_delay_steps):
-            shared_cache[:, idx : idx + tensor_size] = buffered_data.at(read_delay_step)[:, idx : idx + tensor_size]
-            idx += tensor_size
+        tensor_idx = 0
+
+        for sensor_idx, (tensor_size, delay_step, interpolate) in enumerate(
+            zip(
+                shared_metadata.cache_sizes,
+                shared_metadata.delay_steps,
+                interpolate_for_delay or itertools.repeat(False),
+            )
+        ):
+            if interpolate and jitter is not None:
+                shared_cache[:, tensor_idx : tensor_idx + tensor_size] = torch.lerp(
+                    buffered_data.at(delay_step + 1)[:, tensor_idx : tensor_idx + tensor_size],
+                    buffered_data.at(delay_step)[:, tensor_idx : tensor_idx + tensor_size],
+                    jitter[:, sensor_idx].unsqueeze(-1).expand(-1, tensor_size),
+                )
```

Comment:
Linear interpolation is not always valid for interpolating data. The interpolation method should be specified in argument, or as a property of the class, some internal method... whatever. But it should be possible to customise it for the dev implementing their own sensors.